client/src/main/java/io/confluent/kafka/schemaregistry/client/rest/utils/UrlList.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.confluent.kafka.schemaregistry.client.rest.utils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * This class manages a set of urls for accessing an upstream registry. It basically
```

Comment:
You're right. I was thinking the additional urls were primarily for fail over, but we could use them for load-balancing as well.  It's a bit of a slippery slope since that logic can get complex, but it might be worthwhile, especially if users tend to use the same registry configuration across all the hosts that use it.